Exfat fails to build

Exfat DKMs fails to build 
```

DKMS make.log for exfat-1.2.8 for kernel 4.18.10_1 (x86_64)
jeu. sept. 27 20:52:51 CEST 2018
make -C /lib/modules/4.18.10_1/build M=/var/lib/dkms/exfat/1.2.8/build modules
make[1] : on entre dans le répertoire « /usr/src/kernel-headers-4.18.10_1 »
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_core.o
  CC [M]  /var/lib/dkms/exfat/1.2.8/build/exfat_super.o
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c: In function 'exfat_fill_inode':
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:1927:27: error: passing argument 2 of 'exfat_time_fat2unix' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_fat2unix(sbi, &inode->i_mtime, &info.ModifyTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:150:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:1928:27: error: passing argument 2 of 'exfat_time_fat2unix' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_fat2unix(sbi, &inode->i_ctime, &info.CreateTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:150:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:1929:27: error: passing argument 2 of 'exfat_time_fat2unix' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_fat2unix(sbi, &inode->i_atime, &info.AccessTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:150:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c: In function 'exfat_write_inode':
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:2010:27: error: passing argument 2 of 'exfat_time_unix2fat' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_unix2fat(sbi, &inode->i_mtime, &info.ModifyTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:169:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:2011:27: error: passing argument 2 of 'exfat_time_unix2fat' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_unix2fat(sbi, &inode->i_ctime, &info.CreateTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:169:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:2012:27: error: passing argument 2 of 'exfat_time_unix2fat' from incompatible pointer type [-Werror=incompatible-pointer-types]
  exfat_time_unix2fat(sbi, &inode->i_atime, &info.AccessTimestamp);
                           ^~~~~~~~~~~~~~~
/var/lib/dkms/exfat/1.2.8/build/exfat_super.c:169:70: note: expected 'struct timespec *' but argument is of type 'struct timespec64 *'
 void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec *ts,
                                                     ~~~~~~~~~~~~~~~~~^~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:318: /var/lib/dkms/exfat/1.2.8/build/exfat_super.o] Error 1
make[1]: *** [Makefile:1517: _module_/var/lib/dkms/exfat/1.2.8/build] Error 2
make[1] : on quitte le répertoire « /usr/src/kernel-headers-4.18.10_1 »
make: *** [Makefile:35: all] Error 2
```

Thanks

--- a/srcpkgs/exfat-dkms/patches/linux4.18.patch
+++ b/srcpkgs/exfat-dkms/patches/linux4.18.patch
@@ -1,0 +1,33 @@
+diff --git exfat_super.c exfat_super.c
+index 312de36..262b74d 100644
+--- exfat_super.c
++++ exfat_super.c
+@@ -97,6 +97,11 @@ static int exfat_default_codepage = CONFIG_EXFAT_DEFAULT_CODEPAGE;
+ static char exfat_default_iocharset[] = CONFIG_EXFAT_DEFAULT_IOCHARSET;
+ 
+ extern struct timezone sys_tz;
++#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
++#define timespec_compat timespec64
++#else
++#define timespec_compat timespec
++#endif
+ 
+ #if LINUX_VERSION_CODE < KERNEL_VERSION(4,8,0)
+ #define current_time(x)        (CURRENT_TIME_SEC)
+@@ -147,7 +152,7 @@ static time_t accum_days_in_year[] = {
+ static void _exfat_truncate(struct inode *inode, loff_t old_size);
+ 
+ /* Convert a FAT time/date pair to a UNIX date (seconds since 1 1 70). */
+-void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec *ts,
++void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec_compat *ts,
+                                                 DATE_TIME_T *tp)
+ {
+        time_t year = tp->Year;
+@@ -166,7 +171,7 @@ void exfat_time_fat2unix(struct exfat_sb_info *sbi, struct timespec *ts,
+ }
+ 
+ /* Convert linear UNIX date to a FAT time/date pair. */
+-void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec *ts,
++void exfat_time_unix2fat(struct exfat_sb_info *sbi, struct timespec_compat *ts,
+                                                 DATE_TIME_T *tp)
+ {

--- a/srcpkgs/exfat-dkms/template
+++ b/srcpkgs/exfat-dkms/template
@@ -1,7 +1,7 @@
 # Template file for 'exfat-dkms'
 pkgname=exfat-dkms
 version=1.2.8
-revision=4
+revision=5
 _commit=01c30ad52625a7261e1b0d874553b6ca7af25966
 wrksrc="exfat-nofuse-${_commit}"
 short_desc="Exfat kernel driver (nofuse)"

